packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id]/entities.page.tsx

```diff
@@ -0,0 +1,167 @@
+import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
+import { GridCellKind } from "@glideapps/glide-data-grid";
+import {
+  Chip,
+  FontAwesomeIcon,
+  IconButton,
+} from "@hashintel/hash-design-system";
+import { Box, Paper, Stack } from "@mui/material";
+import { useMemo, useState } from "react";
+import { faCircleQuestion } from "@fortawesome/free-regular-svg-icons";
+import { GlideGrid } from "../../../../../components/GlideGlid/glide-grid";
+import { GlideGridOverlayPortal } from "../../../../../components/GlideGlid/glide-grid-overlay-portal";
+import {
+  createHandleHeaderClicked,
+  sortRowData,
+  TableSort,
+} from "../../../../../components/GlideGlid/utils/sorting";
+import { useDrawHeader } from "../../../../../components/GlideGlid/utils/use-draw-header";
+import { SectionWrapper } from "../../../shared/section-wrapper";
+import { WhiteChip } from "../../../shared/white-chip";
+import { blankCell } from "../../../../../components/GlideGlid/utils";
+import { HomeIcon } from "../../../../../shared/icons/home-icon";
+import { EarthIcon } from "../../../../../shared/icons/earth-icon";
+import { renderValueIconCell } from "../value-icon-cell";
+import { useRouteNamespace } from "../use-route-namespace";
+import { useEntitiesTable } from "../use-entities-table";
+import { useEntityTypeEntities } from "../use-entity-type-entities";
+import { NextPageWithLayout } from "../../../../../shared/layout";
+import { getEntityTypeEditorLayout } from "../entity-type-header";
+
+const Page: NextPageWithLayout = () => {
+  const { entities, entityTypes, propertyTypes, subgraph } =
+    useEntityTypeEntities() ?? {};
+
+  const [showSearch, setShowSearch] = useState(false);
+  const [tableSort, setTableSort] = useState<TableSort<string>>({
+    key: "entity",
+    dir: "desc",
+  });
+
+  const namespace = useRouteNamespace();
+
+  const { columns, rows } =
+    useEntitiesTable(entities, entityTypes, propertyTypes, subgraph) ?? {};
+
+  const entitiesCount = useMemo(() => {
+    const namespaceEntities =
+      entities?.filter((entity) => entity.ownedById === namespace?.id) ?? [];
+
+    return {
+      namespace: namespaceEntities.length,
+      public: (entities?.length ?? 0) - namespaceEntities.length,
+    };
+  }, [entities, namespace]);
+
+  const drawHeader = useDrawHeader(tableSort, columns ?? []);
+
+  if (!columns || !rows) {
+    return null;
+  }
+
+  const handleHeaderClicked = createHandleHeaderClicked(
+    columns,
+    tableSort,
+    setTableSort,
+  );
+
+  const sortedRows = rows && sortRowData(rows, tableSort);
```

Comment:
`!rows` has a check above
```suggestion
  const sortedRows = sortRowData(rows, tableSort);
```

Reply:
Removed the redundant check in b6f3b30fbf56c6ca9a4e002ee5da8e3f0c963c2e.